Added documentation for feature/escaping-unicode-control-chars (#771)
arangodb/docs: 3.9/programs-arangod-log.md

## Escaping

There are two flags for retaining or escaping control and unicode 
characters in the log. The flag `--log.escape` is deprecated since v3.9.0, and instead, 
the new flags `--log.escape-control-chars` and `log.escape-unicode-chars` 
should be used. 

- `--log.escape-control-chars`: this flag applies to the control characters, which have hex codes below `\x20`, 
  and also the character DEL with hex code `\x7f`. When the flag value is set to `false`, control 

> characters will be retained when they have a visible representation, and replaced
> with a space character in case they don't have a visible representation. For example, 
> the control characer `\n` is visible, so a `\n` will be displayed in the log. Contrary, 
> the control character `BEL` is not visible, so a space will be displayed instead.
> When the flag value is set to `true`, the hex code for the character is displayed, for 
> example, the `BEL` character will be displayed as its hex code, `\x07`.
> The default value for this flag is `true` to ensure compatibility with 
> previous versions.

Reviewer: ```suggestion
  characters will be retained when they have a visible representation, and replaced
  with a space character in case they don't have a visible representation. For example, 
  the control characer `\n` is visible, so a `\n` will be displayed in the log. Contrary, 
  the control character `BEL` is not visible, so a space will be displayed instead.
  When the flag value is set to `true`, the hex code for the character is displayed, for 
  example, the `BEL` character will be displayed as its hex code, `\x07`.
  The default value for this flag is `true` to ensure compatibility with 
  previous versions.
```

## Escaping

There are two flags for retaining or escaping control and unicode 

> characters in the log. The flag `log.escape` is deprecated and, instead, 

Reviewer: ```suggestion
characters in the log. The flag `--log.escape` is deprecated since v3.9.0, and instead, 
```

the new flags `--log.escape-control-chars` and `log.escape-unicode-chars` 
should be used. 

> - `--log.escape-control-chars`: this flag applies to the control characters, which have hex code below `\x20`, and also the character DEL, with hex code of `\x7f`. When its value is set to false, the control 

Reviewer: ```suggestion
- `--log.escape-control-chars`: this flag applies to the control characters, which have hex codes below `\x20`, 
and also the character DEL with hex code `\x7f`. When the flag value is set to `false`, the control 
```

> character will be retained, and its actual value will be displayed when it 
> is a visible character, or a space ` ` character will be displayed if it is 
> not a visible character. The same will happen to `DEL` character (code `\xF7`), 

Reviewer: ```suggestion
characters will be replaced by a space character in the log output.The same will happen 
to the `DEL` character (hex code `\x7f`), 
```

> character will be retained, and its actual value will be displayed when it 
> is a visible character, or a space ` ` character will be displayed if it is 
> not a visible character. The same will happen to `DEL` character (code `\xF7`), 
> even though it is not a control character, because it is not visible. For example, 
> control characer `\n` is visible, so a `\n` will be displayed in the log, and 
> control character `BEL` is not visible, so a space ` ` would be displayed.
> When its value is set to true, the hex code for the character is displayed, for 
> example, `BEL` character would be displayed as its hex code, `\x07`.
> 	The default value for this flag is `true` for compatibility with 
> previous versions.

Reviewer: ```suggestion
characters will be retained when they have a visible representation, and replaced
with a space character in case they don't have a visible representation. For example, 
the control characer `\n` is visible, so a `\n` will be displayed in the log. Contrary, 
the control character `BEL` is not visible, so a space will be displayed instead.
When the flag value is set to `true`, the hex code for the character is displayed, for 
example, the `BEL` character will be displayed as its hex code, `\x07`.
The default value for this flag is `true` to ensure compatibility with 
previous versions.
```

> - `--log.escape-unicode-chars`: when its value is set to false, the unicode character

Reviewer: ```suggestion
- `--log.escape-unicode-chars`: when its value is set to `false`, Unicode characters
```

> will be retained, and its actual value will be displayed. For example, `犬` will 
> be displayed as `犬`. When its value is set to true, the character is escaped, and 
> the hex code for the character is displayed. For example, `犬` would be displayed 
> as its hex code, `\u72AC`. 
> 	The default value for this flag is set to `false` for compatibility with 
> previous versions.

Reviewer: ```suggestion
will be retained and written to the log as-is. For example, `犬` will 
be logged as `犬`. When the flag value is set to `true`, any Unicode characters are 
escaped, and the hex codes for all Unicode characters are logged instead. 
For example, `犬` would be logged as its hex code, `\u72AC`. 
The default value for this flag is set to `false` for compatibility with 
previous versions.
```
